app/src/main/java/org/raleighmasjid/iar/DayChangedBroadcastReceiver.kt

```diff
@@ -0,0 +1,34 @@
+package org.raleighmasjid.iar
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import java.text.SimpleDateFormat
+import java.util.*
+
+abstract class DayChangedBroadcastReceiver : BroadcastReceiver() {
+    private var date = Date()
+    private val dateFormat by lazy { SimpleDateFormat("yyMMdd", Locale.getDefault()) }
+
+    override fun onReceive(context: Context, intent: Intent) {
+        val action = intent.action
+        val currentDate = Date()
+
+        if ((action == Intent.ACTION_DATE_CHANGED) && !isSameDay(currentDate)) {
```

Comment:
i dont think we need to do this check unless this intent is firing extremely often. I would rather do more refreshes than less.